client/lib/async-requests/index.js

```diff
@@ -77,14 +77,43 @@ export const getTaxRateLabels = getRequestByIdString(
 	} )
 );
 
+/**
+ * Create a variation name by concatenating each of the variation's
+ * attribute option strings.
+ *
+ * @param {Object} variation - variation returned by the api
+ * @param {Array} variation.attributes - attribute objects, with option property.
+ * @param {string} variation.name - name of variation.
+ * @return {string} - formatted variation name
+ */
+export function getVariationName( { attributes, name } ) {
+	const separator =
+		window.wcSettings.variationTitleAttributesSeparator || ' - ';
```

Comment:
May as well use `import { getSetting } from '@woocommerce/wc-admin-settings';` here.

Reply:
Ah, I must have looked at the wrong code to base this on. We still have some instances of directly accessing `window.wcSettings` - particularly when hydrating values in HoCs.. should we be using `getSetting` everywhere instead?

Reply:
Should `getSetting()` from `@woocommerce/wc-admin-settings` be used in our published packages? It doesn't seem to be in any of them currently..

Reply:
>Should getSetting() from @woocommerce/wc-admin-settings be used in our published packages? It doesn't seem to be in any of them currently..

No, because its not actually a package but a webpack alias instead. So `@woocommerce/wc-admin-settings` outside of wc-admin causes a fatal error.

>should we be using getSetting everywhere instead?

That whole paradigm was a mistake, especially exposing the `setSetting` method. The settings data store is a migration of its usage for items that are mutable. We left the rest because its a large change and they are items not likely to change during usage of wc-admin, ie site title. I think the hydration HOC's should stay using `window` because its clear they are an initial hydration only.